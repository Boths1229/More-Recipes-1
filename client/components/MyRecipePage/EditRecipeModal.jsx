import React, { PropTypes } from 'react';
import ImageUploader from 'react-firebase-image-uploader';
import firebase from 'firebase';
import { Pulse } from 'react-preloading-component';


const propTypes = {
  errorMessage: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  onFocus: PropTypes.func.isRequired,
  onProgress: PropTypes.func.isRequired,
  startUpload: PropTypes.func.isRequired,
  upload: PropTypes.func.isRequired,
  isUploading: PropTypes.bool.isRequired,
  progress: PropTypes.number.isRequired,
  details: PropTypes.shape({
    recipeName: PropTypes.string,
    description: PropTypes.string,
    mealType: PropTypes.string,
    ingredients: PropTypes.string,
    method: PropTypes.string
  }).isRequired
};

const EditRecipeModal = props => (
  <div
    className="modal fade"
    id="editModal"
    tabIndex="-1"
    data-backdrop="static"
    role="dialog"
    aria-labelledby="editModalLabel"
    style={{ display: 'none' }}
    aria-hidden="true"
  >
    <div className="modal-dialog" role="document">
      <div className="modal-content">
        <div className="modal-header black">
          <h5 className="modal-title black" id="editModalLabel">Edit Recipe</h5>
          <button type="button" className="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">Ã—</span>
          </button>
        </div>
        <div className="modal-body">
          {
            props.errorMessage &&
            <div className="alert alert-danger">
              {
                props.errorMessage
              }
            </div>
          }
          <form
            onSubmit={props.handleSubmit}
            id="editRecipeForm"
          >
            <div className="form-group">
              <label htmlFor="recipient-name" className="col-form-label black">Title:</label>
              <input
                value={props.details.recipeName}
                onChange={props.handleChange}
                onFocus={props.onFocus}
                type="text"
                name="recipeName"
                className="form-control"
                id="recipient-name"
                required
              />
            </div>
            <div className="form-group">
              <label
                htmlFor="message-text"
                className="col-form-label black"
              >
                Description:
              </label>
              <textarea
                value={props.details.description}
                className="form-control"
                onChange={props.handleChange}
                onFocus={props.onFocus}
                id="message-text"
                name="description"
                required
              />
            </div>
            <div className="form-group">
              <label
                htmlFor="image"
                className="col-form-label black"
              >
                Image:
              </label>
              {
                props.isUploading &&
                props.progress < 100 &&
                <div>
                  <Pulse />
                </div>
              }
              <ImageUploader
                name="image"
                storageRef={
                  firebase
                    .storage()
                    .ref('images')
                }
                onProgress={props.onProgress}
                onUploadSuccess={props.upload}
                onUploadStart={props.startUpload}
              />
            </div>
            <div className="form-group">
              <label
                htmlFor="type-text"
                className="col-form-label black"
              >
                Type of meal:
              </label>
              <select
                value={props.details.mealType}
                onChange={props.handleChange}
                onFocus={props.onFocus}
                name="mealType"
                className="form-control"
                id="type-text"
                required
              >
                <option value="breakfast">Breakfast</option>
                <option value="brunch">Brunch</option>
                <option value="elevenses">Elevenses</option>
                <option value="lunch">Lunch</option>
                <option value="tea">Tea</option>
                <option value="supper">Supper</option>
                <option value="dinner">Dinner</option>
              </select>
            </div>
            <div className="form-group">
              <label
                htmlFor="ingredients-text"
                className="col-form-label black"
              >
                ingredients:
              </label>
              <textarea
                value={props.details.ingredients}
                onChange={props.handleChange}
                onFocus={props.onFocus}
                placeholder="separate each ingredient with a comma"
                name="ingredients"
                className="form-control"
                id="ingredients"
                required
              />
            </div>
            <div className="form-group">
              <label
                htmlFor="method-text"
                className="col-form-label black"
              >
                Method of cooking:
              </label>
              <textarea
                name="method"
                value={props.details.method}
                onChange={props.handleChange}
                onFocus={props.onFocus}
                required
                className="form-control"
                id="method-text"
              />
            </div>
            <div className="modal-footer">
              <button
                type="button"
                className="btn btn-secondary"
                data-dismiss="modal"
              >
                Close
              </button>
              <button
                id="editButton"
                type="submit"
                className="btn vote-button"
              >
                Update Recipe
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
);

EditRecipeModal.propTypes = propTypes;

export default EditRecipeModal;
