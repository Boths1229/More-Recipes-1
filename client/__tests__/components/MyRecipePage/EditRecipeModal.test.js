import React from 'react';
import { shallow } from 'enzyme';
import EditRecipeModal from '../../../components/MyRecipePage/EditRecipeModal';
import MockData from '../../__mocks__/actions/recipes';


let props;

const setup = () => {
  props = {
    errorMessage: '',
    handleSubmit: () => {},
    handleChange: () => {},
    onFocus: () => {},
    onProgress: () => {},
    startUpload: () => {},
    upload: () => {},
    isUploading: false,
    progress: 0,
    details: MockData.recipeData,
    storageRef: {
      firebase: {
        storage: jest.fn()
      }
    }
  };

  return shallow(<EditRecipeModal {...props} />);
};


describe('EditRecipeModal Component', () => {
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('form').length).toBe(1);
    expect(wrapper.find('h5').text()).toBe('Edit Recipe');
  });

  it('should render error if error exists', () => {
    const wrapper = setup();
    wrapper.setProps({ errorMessage: 'Field cannot be empty' });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('form').length).toBe(1);
    expect(wrapper.find('div.alert').text()).toBe('Field cannot be empty');
  });

  it('should render loader image is still uploading', () => {
    const wrapper = setup();
    wrapper.setProps({
      isUploading: true,
      progress: 70
    });
    expect(wrapper).toMatchSnapshot();
  });
});
