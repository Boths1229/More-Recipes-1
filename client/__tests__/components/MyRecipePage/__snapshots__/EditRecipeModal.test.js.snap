// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRecipeModal Component should render correctly 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="editModalLabel"
  className="modal fade"
  data-backdrop="static"
  id="editModal"
  role="dialog"
  style={
    Object {
      "display": "none",
    }
  }
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header black"
      >
        <h5
          className="modal-title black"
          id="editModalLabel"
        >
          Edit Recipe
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <form
          id="editRecipeForm"
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="recipient-name"
            >
              Title:
            </label>
            <input
              className="form-control"
              id="recipient-name"
              name="recipeName"
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value="new recipe"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="message-text"
            >
              Description:
            </label>
            <textarea
              className="form-control"
              id="message-text"
              name="description"
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              value="amazing"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="image"
            >
              Image:
            </label>
            <[object Object]
              name="image"
              onProgress={[Function]}
              onUploadStart={[Function]}
              onUploadSuccess={[Function]}
              storageRef={
                Object {
                  "ref": [MockFunction] {
                    "calls": Array [
                      Array [
                        "images",
                      ],
                    ],
                  },
                  "storage": [MockFunction] {
                    "calls": Array [
                      Array [],
                    ],
                  },
                }
              }
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="type-text"
            >
              Type of meal:
            </label>
            <select
              className="form-control"
              id="type-text"
              name="mealType"
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              value="breakfast"
            >
              <option
                value="breakfast"
              >
                Breakfast
              </option>
              <option
                value="brunch"
              >
                Brunch
              </option>
              <option
                value="elevenses"
              >
                Elevenses
              </option>
              <option
                value="lunch"
              >
                Lunch
              </option>
              <option
                value="tea"
              >
                Tea
              </option>
              <option
                value="supper"
              >
                Supper
              </option>
              <option
                value="dinner"
              >
                Dinner
              </option>
            </select>
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="ingredients-text"
            >
              ingredients:
            </label>
            <textarea
              className="form-control"
              id="ingredients"
              name="ingredients"
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="separate each ingredient with a comma"
              required={true}
              value="awesome"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="method-text"
            >
              Method of cooking:
            </label>
            <textarea
              className="form-control"
              id="method-text"
              name="method"
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              value="great"
            />
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-secondary"
              data-dismiss="modal"
              type="button"
            >
              Close
            </button>
            <button
              className="btn vote-button"
              type="submit"
            >
              Update Recipe
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`EditRecipeModal Component should render error if error exists 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="editModalLabel"
  className="modal fade"
  data-backdrop="static"
  id="editModal"
  role="dialog"
  style={
    Object {
      "display": "none",
    }
  }
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header black"
      >
        <h5
          className="modal-title black"
          id="editModalLabel"
        >
          Edit Recipe
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <div
          className="alert alert-danger"
        >
          Field cannot be empty
        </div>
        <form
          id="editRecipeForm"
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="recipient-name"
            >
              Title:
            </label>
            <input
              className="form-control"
              id="recipient-name"
              name="recipeName"
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value="new recipe"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="message-text"
            >
              Description:
            </label>
            <textarea
              className="form-control"
              id="message-text"
              name="description"
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              value="amazing"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="image"
            >
              Image:
            </label>
            <[object Object]
              name="image"
              onProgress={[Function]}
              onUploadStart={[Function]}
              onUploadSuccess={[Function]}
              storageRef={
                Object {
                  "ref": [MockFunction] {
                    "calls": Array [
                      Array [
                        "images",
                      ],
                      Array [
                        "images",
                      ],
                      Array [
                        "images",
                      ],
                    ],
                  },
                  "storage": [MockFunction] {
                    "calls": Array [
                      Array [],
                      Array [],
                      Array [],
                    ],
                  },
                }
              }
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="type-text"
            >
              Type of meal:
            </label>
            <select
              className="form-control"
              id="type-text"
              name="mealType"
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              value="breakfast"
            >
              <option
                value="breakfast"
              >
                Breakfast
              </option>
              <option
                value="brunch"
              >
                Brunch
              </option>
              <option
                value="elevenses"
              >
                Elevenses
              </option>
              <option
                value="lunch"
              >
                Lunch
              </option>
              <option
                value="tea"
              >
                Tea
              </option>
              <option
                value="supper"
              >
                Supper
              </option>
              <option
                value="dinner"
              >
                Dinner
              </option>
            </select>
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="ingredients-text"
            >
              ingredients:
            </label>
            <textarea
              className="form-control"
              id="ingredients"
              name="ingredients"
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="separate each ingredient with a comma"
              required={true}
              value="awesome"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="method-text"
            >
              Method of cooking:
            </label>
            <textarea
              className="form-control"
              id="method-text"
              name="method"
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              value="great"
            />
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-secondary"
              data-dismiss="modal"
              type="button"
            >
              Close
            </button>
            <button
              className="btn vote-button"
              type="submit"
            >
              Update Recipe
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`EditRecipeModal Component should render loader image is still uploading 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="editModalLabel"
  className="modal fade"
  data-backdrop="static"
  id="editModal"
  role="dialog"
  style={
    Object {
      "display": "none",
    }
  }
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header black"
      >
        <h5
          className="modal-title black"
          id="editModalLabel"
        >
          Edit Recipe
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <form
          id="editRecipeForm"
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="recipient-name"
            >
              Title:
            </label>
            <input
              className="form-control"
              id="recipient-name"
              name="recipeName"
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value="new recipe"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="message-text"
            >
              Description:
            </label>
            <textarea
              className="form-control"
              id="message-text"
              name="description"
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              value="amazing"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="image"
            >
              Image:
            </label>
            <div>
              <n
                color="#FF4601"
                size={10}
              />
            </div>
            <[object Object]
              name="image"
              onProgress={[Function]}
              onUploadStart={[Function]}
              onUploadSuccess={[Function]}
              storageRef={
                Object {
                  "ref": [MockFunction] {
                    "calls": Array [
                      Array [
                        "images",
                      ],
                      Array [
                        "images",
                      ],
                      Array [
                        "images",
                      ],
                      Array [
                        "images",
                      ],
                      Array [
                        "images",
                      ],
                    ],
                  },
                  "storage": [MockFunction] {
                    "calls": Array [
                      Array [],
                      Array [],
                      Array [],
                      Array [],
                      Array [],
                    ],
                  },
                }
              }
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="type-text"
            >
              Type of meal:
            </label>
            <select
              className="form-control"
              id="type-text"
              name="mealType"
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              value="breakfast"
            >
              <option
                value="breakfast"
              >
                Breakfast
              </option>
              <option
                value="brunch"
              >
                Brunch
              </option>
              <option
                value="elevenses"
              >
                Elevenses
              </option>
              <option
                value="lunch"
              >
                Lunch
              </option>
              <option
                value="tea"
              >
                Tea
              </option>
              <option
                value="supper"
              >
                Supper
              </option>
              <option
                value="dinner"
              >
                Dinner
              </option>
            </select>
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="ingredients-text"
            >
              ingredients:
            </label>
            <textarea
              className="form-control"
              id="ingredients"
              name="ingredients"
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="separate each ingredient with a comma"
              required={true}
              value="awesome"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="method-text"
            >
              Method of cooking:
            </label>
            <textarea
              className="form-control"
              id="method-text"
              name="method"
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              value="great"
            />
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-secondary"
              data-dismiss="modal"
              type="button"
            >
              Close
            </button>
            <button
              className="btn vote-button"
              type="submit"
            >
              Update Recipe
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
